lib/guide.md

```diff
@@ -0,0 +1,1061 @@
+
+public class Main {
+    public static void main(String[] args) {
+        // Create a provider for interacting with Starknet
+        Provider provider = new JsonRpcProvider("https://example-node-url.com/rpc");
+        
+        // Set up an account
+        Felt privateKey = Felt.fromHex("0x123");
+        Felt accountAddress = Felt.fromHex("0x1236789");
+        // ⚠️ WARNING ⚠️ Both the account address and key are for demonstration purposes only.
+
+        StarknetChainId chainId = provider.getChainId().send();
+        Account account = new StandardAccount(accountAddress, privateKey, provider, chainId, Felt.ZERO);
+
+        // Specify the contract address, in this example ETH ERC20 contract is used
+        Felt contractAddress = Felt.fromHex("0x049d36570d4e46f48e99674bd3fcc84644ddd6b96f7c741b1562b82f9e004dc7");
+
+        // Create a call
+        List<Felt> calldata = List.of(account.getAddress());
+        Call call = new Call(contractAddress, "balanceOf", calldata);
+        Request<FeltArray> request = provider.callContract(call);
+
+        // Send the call request
+        List<Felt> response = request.send();
+
+        // Output value's type is Uint256 and is represented by two Felt values
+        Uint256 balance = new Uint256(response.get(0), response.get(1));
+
+        System.out.println("Balance: " + balance.toString() + ".");
+    }
+}
+```
+
+## Making multiple calls: get multiple transactions data
+<!-- codeSection(path="starknet/provider/ProviderTest.kt", function="batchGetTransactions", language="kotlin") -->
+```java
+import com.swmansion.starknet.data.types.*;
+import com.swmansion.starknet.data.types.RequestResult;
+import com.swmansion.starknet.provider.rpc.JsonRpcProvider;
+import com.swmansion.starknet.service.http.requests.HttpBatchRequest;
+
+import java.util.List;
+
+import static com.swmansion.starknet.data.Selector.selectorFromName;
+
+public class Main {
+    public static void main(String[] args) {
+        // Create a provider for interacting with Starknet
+        JsonRpcProvider provider = new JsonRpcProvider("https://example-node-url.com/rpc");
+
+        // Batch any RPC requests
+        // Get block hash and number + Check the initial value of `balance` in contract
+        HttpBatchRequest mixedRequest = provider.batchRequestsAny(
+                provider.getBlockHashAndNumber(),
+                provider.getStorageAt(Felt.fromHex("0x123"), selectorFromName("balance"))
+        );
+
+        // Create and send the batch request
+        List<RequestResult> mixedResponse = mixedRequest.send();
+
+        GetBlockHashAndNumberResponse blockHashAndNumber = (GetBlockHashAndNumberResponse) mixedResponse.get(0).getOrNull();
+        Felt contractBalance = (Felt) mixedResponse.get(1).getOrNull();
+
+        System.out.println("Block hash: " + blockHashAndNumber.getBlockHash() + ".");
+        System.out.println("Initial contract balance: " + contractBalance.getValue() + ".");
+    }
+}
+```
```

Comment:
This seems to be a duplicate of "Making multiple calls of different types in one request", and I think the latter makes for a better name 👍🏻

Reply:
Actually the test for mixed requests was shown in example of batching requests of same type - I fixed it and kept both of them.